reo/validators.py

```diff
@@ -1162,18 +1161,21 @@ def check_special_cases(self, object_name_path, template_values=None, real_value
                     # check if user intended to run CHP and supplied sufficient pararmeters to run CHP
                     if user_supplied_chp_inputs:
                         required_keys = prime_mover_defaults_all['recip_engine'].keys()
-                        filtered_values = {k: real_values.get(k) for k in required_keys}
+                        if real_values.get('prime_mover') is None:
+                            self.input_data_errors.append('No prime_mover was input so all cost and performance parameters must be input.')
```

Comment:
@Bill-Becker does this not cover your comment below?

Reply:
@Bill-Becker what happens is that it first checks to see that prime_mover is set, if not an error is added to the error array. It doesn't error out though until checking to see what defaults are missing and adding a message about that. The final error message is two sentences: 

"error": "No prime_mover was input so all cost and performance parameters must be input. CHP is missing a value for the following defaults: attr1, attr2, attr3"

Reply:
So if it doesn't find that any of the required inputs are missing, it won't ever error out? It still seems like it will "error" with the "No prime_mover was input so all cost and performance parameters must be input.'" message because that was appended to the self.input_data_errors message.

Reply:
@Bill-Becker nope, appending to the input_data_errors list doesn't immediately trigger an error. That's actually why we first starting looking into this issue. The original problem was that we were hitting L1178 self.validate_chp_inputs when errors had been found and the job was actually in an invalid format. When errors are found they are actually raised in reo/api.py not validators.py

Reply:
@t-kwasnik sorry for my confusion on this, but you said it doesn't **immediately** trigger an error, but are you also saying that if it doesn't find any missing parameters that it will **never** trigger an error? I am going to make a few tweaks on this branch this morning to reduce the number of required inputs for CHP in the case they don't send a prime_mover type - all of this checking will still be needed, but there will be a lesser number of params to pass.